### PR TITLE
CF flow package for O2

### PR DESCRIPTION
**Pull request by EmilGorm**

### CODE REVIEW

File: PWGCF/GenericFramework/FlowContainer.cxx [hunk 1]
```diff
@@ -0,0 +1,1204 @@
+/*
```
Comment (jgrosseo) [inline PWGCF/GenericFramework/FlowContainer.cxx:1]: Please add the copyright (all files please)

File: PWGCF/Tasks/CMakeLists.txt [hunk 2]
```diff
@@ -29,3 +29,7 @@ o2physics_add_dpl_workflow(correlations
                     PUBLIC_LINK_LIBRARIES O2::Framework O2Physics::AnalysisCore O2Physics::PWGCFCore
                     COMPONENT_NAME Analysis)
 
+o2physics_add_dpl_workflow(flow-generic-framework
+                    SOURCES GenericFramework.cxx
```
Comment (jgrosseo) [inline PWGCF/Tasks/CMakeLists.txt:33]: Start with lower case.
Call the file like the executable --> flowGenericFramework.cxx

File: PWGCF/Tasks/GenericFramework.cxx [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+    LOGF(INFO, "Tracks for collision: %d | Vertex: %.1f | INT7: %d | V0M: %.1f", tracks.size(), collision.posZ(), collision.sel7(), collision.centV0M());
+    float vtxz = collision.posZ();
+    registry.fill(HIST("hVtxZ"), vtxz);
+
+    fGFW->Clear();
+    const auto centrality = collision.centV0M();
+    float l_Random = fRndm->Rndm();
+    float weff = 1, wacc = 1;
+
+    for (auto track = tracks.begin(); track != tracks.end(); ++track) {
```
Comment (jgrosseo) [inline PWGCF/Tasks/GenericFramework.cxx:163]: You can use `for (auto& track : tracks)`
Note in particular the added `&` avoiding an extra copy

File: PWGCF/Tasks/GenericFramework.cxx [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+      weff = 1. / weff;
+      if (cfg.mAcceptance)
+        wacc = cfg.mAcceptance->GetNUA(track.phi(), track.eta(), vtxz);
+      else
+        wacc = 1;
+
+      fGFW->Fill(track.eta(), 1, track.phi(), wacc * weff, 3);
+    }
+    for (unsigned long int l_ind = 0; l_ind < corrconfigs.size(); l_ind++) {
+      FillFC(corrconfigs.at(l_ind), centrality, l_Random);
```
Comment (jgrosseo) [inline PWGCF/Tasks/GenericFramework.cxx:182]: The identification of the config for each track is very slow. Better invert the logic, have this as the outer loop. Ideally measure which of the two is faster. We can merge this before the optimization, though

**Review by jgrosseo — COMMENTED** [review]
Thanks for the code. Find some comments inline.

**Comment by victor-gonzalez** [comment]
I guess you will need a blank line after the copyright notice and any further comment

**Comment by EmilGorm** [comment]
Yea, at least I have just copy/pasted from the copyright checker, but it still fails

**Comment by EmilGorm** [comment]
I will stop spamming commits now, until I understand why it fails

File: PWGCF/Tasks/CMakeLists.txt [hunk 5]
```diff
@@ -29,3 +29,7 @@ o2physics_add_dpl_workflow(correlations
                     PUBLIC_LINK_LIBRARIES O2::Framework O2Physics::AnalysisCore O2Physics::PWGCFCore
                     COMPONENT_NAME Analysis)
 
+o2physics_add_dpl_workflow(flow-generic-framework
+                    SOURCES flowgenericframework.cxx
```
Comment (jgrosseo) [inline PWGCF/Tasks/CMakeLists.txt:33]: There is nothing wrong with Capitals in the filename, just then not in the executable name. So each capital becomes a dash. I.e.:
flowGenericFramework -> flow-generic-framework
You can google camel case vs snake case for your amusement :)

File: PWGCF/Tasks/GenericFramework.cxx (hunk 4 above)
Comment (jgrosseo) [reply PWGCF/Tasks/GenericFramework.cxx]: You test this later I assume or did you already try?

**Comment by jgrosseo** [comment]
> @jgrosseo It looks to me that the licence/copyright checker is doing something wrong

Checking...

**Review by jgrosseo — APPROVED** [review]

**Comment by jgrosseo** [comment]
It does not compile on Mac, see the log of the CI...

**Review by jgrosseo — CHANGES_REQUESTED** [review]
Fix Mac compilation

**Review by jgrosseo — APPROVED** [review]

**Pull request merged**